src/pilot/assign_role.py

```diff
@@ -1160,10 +1165,26 @@ def select_os_volume(os_volume_size_gb, ironic_client, drac_client, node_uuid):
 
                     break
 
+            # Note: This code block represents single disk scenario.
             if raid_size_gb == 0:
-                raise RuntimeError("There must be either a virtual disk that "
-                                   "is not a RAID 0 to install the OS on, or "
-                                   "os-volume-size-gb must be specified")
+                try:
+                    physical_disk_view_doc = drac_client.enumerate(
```

Comment:
I was talking with Rick today, and we came up with a case that needs to be handled which requires a change to this logic.  This is the case where the server has a RAID controller that does not support JBOD mode.  In that case, a RAID0 will be created that contains a single disk.  As a result, I think the logic needs to be:
- Check the virtual disks and see if there is a single RAID0.  If so then use that.  If there is more than one RAID0 then raise an exception
- If there are no RAID0's, then check the physical disks and see if there is a single disk.  If so, then use that.  If there is 0 or more than 1 physical disk, then raise an exception.

Reply:
Chris - If there is single disk which is RAID0 and single physical disk without raid, anyway our code is taking the disk size irrespective of the disk type. If this is the case, then is it ok if we remove the RAID0 code for single disk ?
I didn't understand the RAID0 scenario for single disk which you are refering.
May be we can discuss this in today's meeting.